Chapter on Software Checklist (#910)
alan-turing-institute/the-turing-way: book/content/software_checklist/01/checklist_matrix.md

> # [WIP] Checkmatrix software projects

Reviewer: I don't know what WIP means. Is that Work In Progress? If so, this should not be in the content.
There is no word 'checkmatrix', maybe use 'checklist matrix' instead?

Author: done

> This matrix shows what parts of the software sustainability checklist should be taken care of at (perhaps slightly before) what state of a project.

Reviewer: ```suggestion
This matrix shows which parts of the software sustainability checklist should be taken care of at (perhaps slightly before) which stage of a project.
```

Author: done

## Explanation of project states

- [*Prototype phase*](checklist/checklist_prototype.md). The first step in most software development is trying out different things with no intention in keeping the intermediate results. Signs you could be in this phase:
  - You switch programming languages.
  - You throw away all of your code once in a while.
  - You work on the code by yourself.
  - You are waiting with showing other people your code until you "clean it up a bit first".

> - [*Pre-release phase*](checklist/checklist_prerelease.md). Eventually you get software you intend to keep. Signs you could be in this phase:

Reviewer: ```suggestion
- [*Pre-release phase*](checklist/checklist_prerelease.md). Eventually, you get the software you intend to keep. Signs you could be in this phase:
```

Author: done

  - You have multiple developers.
  - You have external contributors.
  - You are working up to a release.
  - Users ask you if the software is done yet.

> - [*Maturity phase*](checklist/checklist_mature.md): Software that has reached maturity, has a clear function and scope, and is used. Signs you could be in this phase:

Reviewer: ```suggestion
- [*Maturity phase*](checklist/checklist_mature.md): Software that has reached maturity has a clear function and scope and is used. Signs you could be in this phase:
```

Author: done

  - The software has a release.

>   - The software has users: people actually using your software/code.

Reviewer: ```suggestion
  - The software has users: people using your software/code.
```

Author: done

>   - You have external contributor.

Reviewer: ```suggestion
  - You have an external contributor.
```

Author: done
